src/ai_hawk/job_manager.py

```diff
@@ -454,29 +458,56 @@ def next_job_page(self, position, location, job_page):
         self.driver.get(
             f"https://www.linkedin.com/jobs/search/{self.base_search_url}&keywords={encoded_position}{location}&start={job_page * 25}")
 
-
     def job_tile_to_job(self, job_tile) -> Job:
         logger.debug("Extracting job information from tile")
         job = Job()
 
+        # Extract job Title
         try:
-            job.title = job_tile.find_element(By.CLASS_NAME, 'job-card-list__title').find_element(By.TAG_NAME, 'strong').text
+            title_element = job_tile.find_element(
+                By.XPATH, ".//div[contains(@class, 'artdeco-entity-lockup__title')]//a"
+            )
+            job.title = title_element.text.strip()
             logger.debug(f"Job title extracted: {job.title}")
         except NoSuchElementException:
             logger.warning("Job title is missing.")
-        
+
+        # Extract job Link
         try:
-            job.link = job_tile.find_element(By.CLASS_NAME, 'job-card-list__title').get_attribute('href').split('?')[0]
+            job.link = title_element.get_attribute("href").split("?")[0]
```

Comment:
There’s a potential unhandled exception here. If the previos line raised an exception and `title_element` is not found, this line will fail and raise a different exception rather than NoSuchElementException, causing the entire program to stop abruptly.
Suggestion: `job.link = job_tile.find_element(By.CLASS_NAME, 'job-card-list__title--link').get_attribute('href').split('?')[0]`